### PR TITLE
Updates Zenodo (DOI)

### PR DESCRIPTION
**Pull request by jlaura**
The DOI entered will now always resolve tot he newest release. The other DOI was pinned to a specific release.

This change updates the DOI number and link in the README.

## Description

As above, this is an update so that the DOI always resolves to the latest ISIS version.

## Related Issue

#3810 

## Motivation and Context

Right now we have to manually update the DOI with every release. This automates that process. 
## How Has This Been Tested?

I followed the link and it got me to 4.2.0_RC1 the current latest release.
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [x] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Comment by jlaura** [comment]
@rbeyer Once you accept the invite on the repo - would you mind giving this PR a review / merge if it looks good? Thanks!

**Review by jessemapel — DISMISSED** [review]

**Review by rbeyer — DISMISSED** [review]

**Comment by jlaura** [comment]
Apologies - I dismissed the stale review because I felt this was incomplete as I dig more into the Zenodo documentation. @rbeyer @jessemapel  I wanted to also include a short section to help citing authors pull the correct DOI. I think this is timely because of all the versioning discussion we have been having.

File: README.md [hunk 1]
```diff
@@ -284,6 +285,9 @@ To build and compile ISIS requires following the instructions listed below, whic
 -   [Building ISIS3 documentation](https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#building-isis3-documentation)
 -   [What to do if you encounter any problems](https://github.com/USGS-Astrogeology/ISIS3/wiki/Developing-ISIS3-with-cmake#problems)
 
+## Citing ISIS
+This project uses a [Zenodo](https://zenodo.org) generated DOI. The badge at the top of this README links to the DOI for the [latest release](https://doi.org/10.5281/zenodo.2563341). It is [good practice](https://help.zenodo.org)(See 'Which DOI Should I Use in Citations?) to cite the version of the software being used by the citing work. To obtain this DOI, one can follow the [link to the latest version](https://doi.org/10.5281/zenodo.2563341) and then check the right sidebar area titled **Versions** for a listing of all ISIS versions that currently have a Zenodo DOI.
```
Comment (rbeyer) [inline README.md]: Needs a space between the "good practice" link text and the parenthetical that starts with "(See 'Which ...".

**Review by rbeyer — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jlaura) [reply README.md]: :+1:

Comment (rbeyer) [reply README.md]: Now there needs to be a closing quote on "Which DOI Should I Use in Citations?".

**Review by jessemapel — COMMENTED** [review]
Text looks good after Ross' comments (I am terrible at copy editing so I'm defering to him for that). Looking at this we need to update our badges, we don't use gitter anymore and the conda badge points to the old isis3 package. I'll put in a separate PR for that.

File: README.md (hunk 1 above)
Comment (jlaura) [reply README.md]: 23 hours later - the quote has been successfully added.

**Review by rbeyer — APPROVED** [review]

**Pull request merged**